# Upwork Analysis of Data Projects

I wanted to jump into freelance work with my background in data and programming. There are just a few issues for new freelancers that are just starting out. First, you don't have any ratings to show your worth. Second, what kind of work are potential employers looking for and what is a competitive price for that work.

This analysis seeks to find the correlation between freelancers with high rating, the services they provide and the prices they offer.

In [2]:
import pandas as pd
import nltk
import matplotlib.pyplot as plt
import seaborn as sns


In [6]:
# load the dataset
projects_df = pd.read_csv('./data/upwork_projects.csv', index_col='Unnamed: 0')

projects_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 984 entries, 0 to 983
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   title        984 non-null    object 
 1   price        984 non-null    float64
 2   rating       984 non-null    float64
 3   num_ratings  984 non-null    int64  
 4   url          984 non-null    object 
dtypes: float64(2), int64(1), object(2)
memory usage: 38.4+ KB


984 entries is not a lot. The original search showed around 40,000 projects for the search term 'data'. I believe there might be some type of reset feature when the count gets close to 1000. That said, the assumption is made that we can determine a trend given data we have.

Before jumping into an analysis of the titles lets take a look at the distribution of price, rating and num_ratings.